src/Symfony/Component/Messenger/Bridge/Redis/Transport/RedisReceiver.php

```diff
@@ -38,24 +38,34 @@ public function __construct(Connection $connection, SerializerInterface $seriali
      */
     public function get(): iterable
     {
-        $redisEnvelope = $this->connection->get();
+        $message = $this->connection->get();
+
+        if (\array_key_exists('message', $message)) {
+            $redisEnvelope = json_decode($message['message'], true);
+        } else {
+            $redisEnvelope = $message;
+        }
 
         if (null === $redisEnvelope) {
             return [];
         }
```

Comment:
For me it looks like the `get` returns now `null` or `array{id: string, message: string}` so I think the check for `message` is unnecessary as it always exists if the return value is not null:

```suggestion
        $redisEnvelope = $this->connection->get();
        
        if (null === $message) {
            return [];
        }

        $redisEnvelope = \json_decode($redisEnvelope['message'], true);
```